packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/ArbTokenBridge.sol

```diff
@@ -86,35 +93,70 @@ contract ArbTokenBridge is CloneFactory {
         l1Pair = _l1Pair;
     }
 
+    function mintAndCall(
+        IArbToken token,
+        uint256 amount,
+        address sender,
+        address dest,
+        bytes memory data
+    ) public {
+        // TODO: should this only be callable by mintERC20?
+        require(msg.sender == address(this), "Mint can only be called by self");
+
+        token.bridgeMint(dest, amount, '');
+        bool success = ITransferReceiver(dest).onTokenTransfer(sender, amount, data);
+
+        require(success, "External onTokenTransfer reverted");
+    }
+
     function mintERC777FromL1(
         address l1ERC20,
-        address account,
+        address sender,
+        address dest,
         uint256 amount,
-        uint8 decimals
+        uint8 decimals,
+        bytes calldata callHookData
     ) external onlyEthPair {
         IArbToken token = ensureERC777TokenExists(l1ERC20, decimals);
-        token.bridgeMint(account, amount);
+        token.bridgeMint(dest, amount, callHookData);
```

Comment:
I think we need a try-catch and mint to sender for this probably

Reply:
If the sender is encoded in the callhookdata then they can achieve the same functionality.

I'm a bit hesitant of adding a post-call hook on something that triggers a post transfer hook. It should be alright, but I'm hesitant.

Reply:
Added in